packages/design/Checkbox.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from "react";
+import classNames from "classnames";
+
+import { Badge } from "./Badge";
+
+import styles from "./Checkbox.module.css";
+
+export type CheckboxProps = { label?: React.ReactNode; secondaryLabel?: React.ReactNode } & Pick<
+  React.HTMLProps<HTMLInputElement>,
+  "id" | "checked" | "className" | "disabled" | "onChange" | "title"
+>;
+
+export function Checkbox({
+  id,
+  checked,
+  className,
+  disabled,
+  label,
+  secondaryLabel,
+  onChange,
+  title,
+  ...props
+}: CheckboxProps) {
+  const checkboxProps = {
+    disabled,
+    id,
+    checked,
+    onChange,
+    type: "checkbox",
+  };
+
+  return label ? (
+    <label className={classNames(styles.Label, className)} {...props}>
+      <input {...checkboxProps} className={styles.Input} />
+      <span>{label}</span>
+      {secondaryLabel ? <Badge label={secondaryLabel} /> : null}
+    </label>
+  ) : (
+    <input {...checkboxProps} {...props} className={classNames(styles.Input, className)} />
```

Comment:
IIR, the reason I was using the `htmlFor` attribute was that the text didn't truncate with "..." nicely if the input and display text were both inside of a `<label>`. I assume you've got that sorted out though so cool 😄

Reply:
Actually this looks like a regression.
![Kapture 2022-08-10 at 10 53 32](https://user-images.githubusercontent.com/29597/183935694-8064bfee-9042-4529-96d7-1e6dc1162a53.gif)

Reply:
Oops, yeah I'll get this working again! The span should be able to handle that.